build-system/tasks/check-types.js

```diff
@@ -283,13 +283,20 @@ async function typeCheck(targetName) {
     return;
   }
 
+  if (target.onError) {
+    // If an onError handler is defined, steal the output and log it manually.
+    opts.logger = (m) => {
+      errorMsg = m;
+      log(m);
+    };
+  }
+
   let errorMsg;
```

Comment:
Doesn't this var declaration need to be above the if statement?

Reply:
Need, no (hoisted); should, definitely. Moved.